### PR TITLE
[webgl] modularize exp, linSpace, multinomial, softmax

### PR DESCRIPTION
**Pull request by tafsiri**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4305)

### CODE REVIEW

**Review by annxingyuan — APPROVED** [review]
Thank you Yannick!!!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4305#-:-MNP8xe-5m17uh1-nUL0:b-zdorv0)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @annxingyuan, @lina128, and @tafsiri)

___
*[tfjs-backend-webgl/src/kernels/Softmax.ts, line 61 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4305#-MNP8lZr3QOgKoSf5N2p:-MNP8lZr3QOgKoSf5N2q:bz945l) ([raw file](https://github.com/tensorflow/tfjs/blob/c08778d599e1513af465f34134cdb0698e8e079b/tfjs-backend-webgl/src/kernels/Softmax.ts#L61)):*
> ```XML
>       realDiv({inputs: {a: b, b: sumExpReshaped}, backend}) as TensorInfo;
> 
>   backend.disposeIntermediateTensorInfo(maxLogit);
> ```

should we use the `const intermediates: TensorInfo[] = []; intermediates.push(...);` approach here?

**Review by tafsiri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4305)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @annxingyuan and @lina128)

___
*[tfjs-backend-webgl/src/kernels/Softmax.ts, line 61 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4305#-MNP8lZr3QOgKoSf5N2p:-MNPQulB09GOuHpVgWNS:b-nxpffv) ([raw file](https://github.com/tensorflow/tfjs/blob/c08778d599e1513af465f34134cdb0698e8e079b/tfjs-backend-webgl/src/kernels/Softmax.ts#L61)):*
<details><summary><i>Previously, annxingyuan (Ann Yuan) wrote…</i></summary><blockquote>

should we use the `const intermediates: TensorInfo[] = []; intermediates.push(...);` approach here?
</blockquote></details>

I think either approach is fine. I usually use the array form when there is are multiple return paths or a loop. In this case its pretty linear so i just matched the cpu impl of softmax.
